py/selenium/webdriver/support/expected_conditions.py

```diff
@@ -63,6 +63,55 @@ def __call__(self, driver):
         return _find_element(driver, self.locator)
 
 
+class url_contains(object):
+    """ An expectation for checking that the current url contains a
+    case-sensitive substring.
+    url is the fragment of url expected,
+    returns True when the title matches, False otherwise
+    """
+    def __init__(self, url):
+        self.url = url
+
+    def __call__(self, driver):
+        return self.url in driver.current_url
+
+
+class url_matches(object):
+    """An expectation for checking the current url.
+    pattern is the expected pattern, which must be an exact match
+    returns True if the title matches, false otherwise."""
+    def __init__(self, pattern):
+        self.pattern = pattern
+
+    def __call__(self, driver):
+        import re
+        match = re.match(self.pattern, driver.current_url)
```

Comment:
I would use `re.search` so that you can use any regular expression and not one that has an implicit `^` match at beginning.